### DISCUSSION TITLE
defining a UDF for any index calculation

### DISCUSSION THREAD

**Discussion by dryginmo** [davidfrantz/force#178] [Q&A]
It´s not quite clear to me how to write a UDF to calculate any index that is not available yet. Following the "Medoid" example from the user manual I was able to produce datasets, but while processing the error "name 'inarray' is not defined." popped up. I don´t understand this behavior. Further, how would I access certain bands to calculate any index I want? For the NDVI I have tried to define the bands for sentinel this way. b4 = inarray[:,4,0,0] ; b8 = inarray[:,8,0,0], but was also faced with the "name 'inarray' is not defined." error. Is this the way to access the bands?

Thanks for your inputs.
Johannes

**Comment by davidfrantz** [comment]
Hi Johannes, 

can you please share with us your UDF?

Best,
David
